test/com/google/enterprise/adaptor/opentext/OpentextAdaptorTest.java

```diff
@@ -1942,45 +1897,55 @@ public void testDoMilestone() throws IOException {
-    assertEquals("45", responseMetadata.get("Duration").get(0));
-    assertEquals("2", responseMetadata.get("NumActive").get(0));
-    assertEquals("0", responseMetadata.get("NumCancelled").get(0));
-    assertEquals("3", responseMetadata.get("NumCompleted").get(0));
-    assertEquals("5", responseMetadata.get("NumInProcess").get(0));
-    assertEquals("2", responseMetadata.get("NumIssue").get(0));
-    assertEquals("0", responseMetadata.get("NumLate").get(0));
-    assertEquals("4", responseMetadata.get("NumOnHold").get(0));
-    assertEquals("7", responseMetadata.get("NumPending").get(0));
-    assertEquals("8", responseMetadata.get("NumTasks").get(0));
-    assertEquals("2012-02-01",
-        responseMetadata.get("OriginalTargetDate").get(0));
-    assertEquals("15.0", responseMetadata.get("PercentCancelled").get(0));
-    assertEquals("55.0", responseMetadata.get("PercentComplete").get(0));
-    assertEquals("32.0", responseMetadata.get("PercentInProcess").get(0));
-    assertEquals("11.0", responseMetadata.get("PercentIssue").get(0));
-    assertEquals("4.0", responseMetadata.get("PercentLate").get(0));
-    assertEquals("45.0", responseMetadata.get("PercentOnHold").get(0));
-    assertEquals("13.0", responseMetadata.get("PercentPending").get(0));
-    assertEquals("99", responseMetadata.get("Resources").get(0));
-    assertEquals("2013-02-01", responseMetadata.get("TargetDate").get(0));
-    List<String> anchors = responseMock.getAnchors();
-    assertEquals(3, anchors.size());
+    assertEquals(expected, baos.toString("UTF-8"));
     assertEquals(
-        Lists.newArrayList("Document 1", "Document 2", "Document 3"),
-        anchors);
+        new com.google.enterprise.adaptor.Metadata(
+            new ImmutableMap.Builder<String, String>()
+            .put("ActualDate", "2011-02-01")
+            .put("Duration", "45")
+            .put("NumActive", "2")
+            .put("NumCancelled", "0")
+            .put("NumCompleted", "3")
+            .put("NumInProcess", "5")
+            .put("NumIssue", "2")
+            .put("NumLate", "0")
+            .put("NumOnHold", "4")
+            .put("NumPending", "7")
+            .put("NumTasks", "8")
+            .put("OriginalTargetDate", "2012-02-01")
+            .put("PercentCancelled", "15.0")
+            .put("PercentComplete", "55.0")
+            .put("PercentInProcess", "32.0")
+            .put("PercentIssue", "11.0")
+            .put("PercentLate", "4.0")
+            .put("PercentOnHold", "45.0")
+            .put("PercentPending", "13.0")
+            .put("Resources", "99")
+            .put("TargetDate", "2013-02-01")
+            .build()
+            .entrySet()),
+        response.getMetadata());
+    assertEquals(ImmutableSet.of("Document 1", "Document 2", "Document 3"),
+        getAnchorKeySet(response));
+  }
+
+  /** Returns the keySet of the anchors in the response. */
+  // TODO(bmj): This can go away if we upgrade to guava 19
```

Comment:
It can't go away because you're passing in a RecordingResponse. Maybe pass in an Iterable<Map.Entry> instead? Why can it go away? Also, this is used 5 times, and defined between the first and second uses, which feels weird.

Reply:
It can go away because Guava 19 supports ImmutableMap.copyOf(Iterable<Map.Entry>), so calls to getAnchorKeySet could be replaced with ImmutableMap.copyOf(response.getAnchors()).keySet()